### PR TITLE
Fix provider review handoff evidence retry

### PR DESCRIPTION
**Pull request by Kbediako**
## What
- Resolve provider-worker review handoff from fresh same-issue nested `implementation-gate` review telemetry when parent `review/telemetry.json` is absent.
- Keep governed review semantics fail-closed when clean evidence is missing, stale, unbound, or contract-invalid.
- Suppress manifest-only retry projection for completed successful `issue_review_handoff` proofs.
- Add the CO-560 docs-first packet and focused positive/negative regressions.

## Why
CO-515 reached review handoff cleanly, but the parent provider run became retryable because only the parent run's top-level review telemetry path was checked. This makes the parent converge on the governed review evidence the worker actually used without accepting missing telemetry as clean.

## Validation
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run test -- CommandRunnerReviewEvidenceConsistency SelectedRunProjection`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- Manifest-backed `codex-orchestrator review --uncommitted` with `review_verdict=clean`, contract mode `enforce`, validation `valid`, overall `clean`
- `git diff --check`
- `npm run pack:smoke`
- `co-status --format json --run-dir /Users/kbediako/Code/CO/.runs/local-mcp/cli/control-host` showed `retrying=0`, `retrying_issues=[]`, and `co515_active_or_retrying=false`

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented false retries after successful provider→worker review handoffs when parent telemetry is missing but valid nested evidence exists; completed handoffs are no longer queued as retries.

* **Documentation**
  * Added PRD, tech spec, action plan, task checklist, registry entries and validation notes describing evidence-governance, fail-closed behaviour and rollout/rollback guidance.

* **Tests**
  * Added tests covering nested evidence, retry classification and evidence-binding edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Kbediako/CO/pull/844?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

Diff budget override: CO-560 requires a docs-first packet plus focused provider-worker review/retry regressions; the scoped local hard gate remains within budget and the stacked branch only exceeds by required traceability and test coverage.

### CODE REVIEW

File: orchestrator/src/cli/services/commandRunner.ts [hunk 1]
```diff
@@ -1138,6 +1215,208 @@ async function readReviewTelemetryEvidence(
...
+    identity.issueIdentifier &&
+    manifestIssueIdentifier &&
+    manifestIssueIdentifier !== identity.issueIdentifier
+  ) {
+    return false;
+  }
+  return Boolean(
+    manifestIssueId === identity.issueId ||
+      (identity.issueIdentifier && manifestIssueIdentifier === identity.issueIdentifier) ||
+      identity.taskEntry === `linear-${identity.issueId}`
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/services/commandRunner.ts]: Fixed in 8c0ebec50c9e1f82b049925a94cad80b9e57180d: nested evidence now requires provider-attempt lineage via proof-linked child streams or matching parent_run_id, with unlineaged same-issue evidence covered by a fail-closed regression.

File: orchestrator/src/cli/control/selectedRunProjection.ts [hunk 2]
```diff
@@ -282,14 +282,38 @@ export async function discoverCompatibilityCollectionContexts(
...
   }
 
   return { running, retrying, all };
 }
 
+function isCompatibilityRetryFallbackContext(context: ControlCompatibilitySourceContext): boolean {
+  if (isSucceededProviderReviewHandoffProofContext(context)) {
+    return false;
+  }
+  return context.rawStatus === 'failed' || context.rawStatus === 'cancelled' || context.rawStatus === 'canceled';
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/selectedRunProjection.ts:301]: Fixed in 8c0ebec50c9e1f82b049925a94cad80b9e57180d: retry suppression now requires a succeeded implementation-gate child stream for failed historical handoff manifests, and the paired negative test keeps failed review handoffs retryable when that clean child evidence is absent.

File: orchestrator/src/cli/services/commandRunner.ts [hunk 3]
```diff
@@ -1138,6 +1215,208 @@ async function readReviewTelemetryEvidence(
...
+  proof: Record<string, unknown> | null;
+  proofAttemptStartedAt: string | null;
+}): Promise<ProviderReviewTelemetryEvidence | null> {
+  const workspacePath = coerceTelemetryString(options.proof?.workspace_path);
+  const issueId = coerceTelemetryString(options.proof?.issue_id);
+  if (!workspacePath || !isAbsolute(workspacePath) || !issueId) {
+    return null;
+  }
+  const issueIdentifier = coerceTelemetryString(options.proof?.issue_identifier);
+  const runsRoot = join(workspacePath, '.runs');
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/services/commandRunner.ts]: Fixed in 8c0ebec50c9e1f82b049925a94cad80b9e57180d: nested review discovery now uses only the configured runs root, and `CommandRunnerReviewEvidenceConsistency` includes a custom-runs negative case that rejects stale default-root clean evidence.

File: orchestrator/src/cli/control/selectedRunProjection.ts [hunk 4]
```diff
@@ -282,14 +282,68 @@ export async function discoverCompatibilityCollectionContexts(
...
+      debugWorker.owner_phase === 'ended' &&
+      debugWorker.owner_status === 'succeeded' &&
+      readStringValue(proof, 'end_reason') === 'issue_review_handoff';
+    if (!succeededReviewHandoff) {
+      return false;
+    }
+    if (context.rawStatus !== 'failed') {
+      return true;
+    }
+    return providerProofHasSucceededImplementationGateChildStream(proof, context);
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/selectedRunProjection.ts]: Fixed in f0f50301fd613195c02b3239f875ddd7e6160fe6: retry suppression for failed historical handoff manifests now requires the exact missing top-level telemetry contract phrase plus succeeded implementation-gate proof. I also added a production-shaped negative regression where enforced contract evidence is missing, the summary includes only an expected telemetry path, and the run remains retrying. The final enforce-mode standalone review is clean.

File: orchestrator/tests/CommandRunnerReviewEvidenceConsistency.test.ts [hunk 5]
```diff
@@ -1020,6 +1020,226 @@ describe('runCommandStage review evidence consistency', () => {
...
+      });
+      await writeNestedImplementationGateReviewArtifacts(
+        input,
+        {
+          generated_at: new Date(Date.parse(proofAttemptStartedAt) + 2).toISOString()
+        },
+        {
+          taskId: 'linear-CO-560',
+          omitIssueIdentity: true
+        }
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/tests/CommandRunnerReviewEvidenceConsistency.test.ts:1167]: Fixed in f0f50301fd613195c02b3239f875ddd7e6160fe6: the nested implementation-gate fixture is now lineage-valid via the current provider run id but identity-invalid with issue id lin-issue-560 / CO-560, so the regression specifically covers issue-binding relaxation. Focused tests passed with 186 cases.
